api/serializers/enrollment_test.py

```diff
@@ -50,3 +52,16 @@ class Meta:
 
     # TODO Think about some validation, e.g. answers should be unique for each question
     #  and should be in options for this question
+
+
+class EnrollmentTestResultLevelSerializer(serializers.ModelSerializer[EnrollmentTestResult]):
+    """A serializer used to get level of language based on how many answers are correct."""
+
+    @staticmethod
+    def calculate_level(answer_ids: Sequence[int]) -> dict[str, str]:
```

Comment:
I think it should be SerializerMethodField. This field will be read-only, and `answers` should be write-only, so in view you will be able to general staff like `validate` and then `return serializer.data`

Reply:
Fixed, I think. Wow that took some effort :)